src/main/java/gift/Service/CategoryService.java

```diff
@@ -0,0 +1,57 @@
+package gift.Service;
+
+import gift.Model.Category;
+import gift.Model.RequestCategory;
+import gift.Model.ResponseCategoryDTO;
+import gift.Repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+
+@Service
+public class CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryService(CategoryRepository categoryRepository){
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Transactional
+    public Category addCategory(RequestCategory requestCategory) {
+        Category category = new Category(requestCategory.name(), requestCategory.color(), requestCategory.imageUrl(), requestCategory.description());
+        return categoryRepository.save(category);
+    }
+
+    public List<ResponseCategoryDTO> getCategories() {
+        List<Category> categoryList = categoryRepository.findAll();
+        List<ResponseCategoryDTO> responseCategoryDTOList = new ArrayList<>();
+        for(Category category : categoryList){
+            responseCategoryDTOList.add(new ResponseCategoryDTO(
+                    category.getId(),
+                    category.getName(),
+                    category.getColor(),
+                    category.getImageUrl(),
+                    category.getDescription()
+            ));
+        }
+        return responseCategoryDTOList;
```

Comment:
java의 stream을 사용하시면 아래 코드처럼도 짜볼 수 있어요 😄 꼭 반영하지 않으셔도되고 자주 사용되니 참고로 알아두세요.
```suggestion
        return categoryRepository.findAll()
                .stream()
                .map(it -> new ResponseCategoryDTO(
                        it.getId(),
                        it.getName(),
                        it.getColor(),
                        it.getImageUrl(),
                        it.getDescription()))
                .toList();
```

Reply:
감사합니다 해당사항 반영토록 하겠습니다.